ocs_ci/deployment/acm.py

```diff
@@ -74,15 +74,27 @@ def __init__(self):
     def deploy(self):
         if self.source == "upstream":
             self.deploy_upstream()
-        else:
+        elif self.source == "downstream":
             self.deploy_downstream()
+        else:
+            raise Exception(f"The Submariner source: {self.source} is not recognized")
 
     def deploy_upstream(self):
         self.download_binary()
         self.submariner_configure_upstream()
 
     def deploy_downstream(self):
-        raise NotImplementedError("Deploy downstream functionality not implemented")
+        if config.DEPLOYMENT.get("ui_deployment"):
```

Comment:
For now I would go with removing this configuration value.

Once someone will implement UI deployment for upstream or CLI deployment for downstream I would start thinking and adding the logic based on config value. But as we don't have the UI deployment for upstream, and here we don't have CLI for downstream let's remove this condition and config value from config file.

Only once someone will implement other logic we will add some logic for it.